CommandConsole.cpp

```diff
@@ -420,3 +441,74 @@ HOOK_METHOD(MouseControl, OnRender, () -> void)
     PrintHelper::GetInstance()->Render();
     super();
 }
+
+HOOK_METHOD(InputBox, StartInput, () -> void)
+{
+    LOG_HOOK("HOOK_METHOD -> InputBox::StartInput -> Begin (CommandConsole.cpp)\n")
+    CommandConsole::GetInstance()->cursorPosition = 0;
+    super();
+}
+
+HOOK_METHOD(InputBox, TextEvent, (CEvent::TextEvent event) -> void)
+{
+    LOG_HOOK("HOOK_METHOD -> InputBox::TextEvent -> Begin (CommandConsole.cpp)\n")
+
+    hs_log_file("TextEvent Cursor position before: %d\n", CommandConsole::GetInstance()->cursorPosition);
+    hs_log_file("TextEvent Event: %d\n", event);
+    cursorTickCount = 0;
+    if (event == 3 && CommandConsole::GetInstance()->cursorPosition > 0)
+    {
+        CommandConsole::GetInstance()->cursorPosition--;
+        inputText.erase(CommandConsole::GetInstance()->cursorPosition, 1);
+        return;
+    }
+    if (event == 4 && CommandConsole::GetInstance()->cursorPosition < inputText.length())
+    {
+        inputText.erase(CommandConsole::GetInstance()->cursorPosition, 1);
+        return;
+    }
+    if (event == 5 && CommandConsole::GetInstance()->cursorPosition > 0) CommandConsole::GetInstance()->cursorPosition--;
+    if (event == 6 && CommandConsole::GetInstance()->cursorPosition < inputText.length() ) CommandConsole::GetInstance()->cursorPosition++;
+    hs_log_file("TextEvent Cursor position after: %d\n", CommandConsole::GetInstance()->cursorPosition);
+
+    super(event);
+}
+
+HOOK_METHOD(InputBox, TextInput, (int ch) -> void)
+{
+    LOG_HOOK("HOOK_METHOD -> InputBox::TextInput -> Begin (CommandConsole.cpp)\n")
+
+    CommandConsole::GetInstance()->InputData(G_->GetWorld()->commandGui, ch);
+}
+
+HOOK_METHOD(InputBox, OnRender, () -> void)
+{
+    LOG_HOOK("HOOK_METHOD -> InputBox::OnRender -> Begin (CommandConsole.cpp)\n")
+        
+    if (bOpen == false) return;
+
+    Point *pos = new Point(0x146, 0xd2);
+    textBox->Draw(pos->x, pos->y);
+
+    pos->y = pos->y + 0x37;
+    pos->x = pos->x + 0x19;
+
+    size_t cursorPosition = CommandConsole::GetInstance()->cursorPosition;
+    if (cursorPosition > inputText.length())
+    { 
+        cursorPosition = inputText.length(); 
+        CommandConsole::GetInstance()->cursorPosition = cursorPosition;
+    }
+
+    std::string commandText = inputText;
+    int inputTextCursorPosition = freetype::easy_measureWidth(8, inputText.substr(0, cursorPosition));
+
+    Pointf posMain = freetype::easy_printAutoNewlines(8,(float)pos->x,(float)pos->y,0x1ea,mainText);
+    Pointf posInput = freetype::easy_printAutoNewlines(8,(float)pos->x, posMain.y + 10.0, 0x1ea, inputText);
+
+    if (cursorTickCount++ < 50) freetype::easy_printAutoNewlines(10,pos->x + (inputTextCursorPosition % 0x1ea) - 1.5, posInput.y-13.5, 0x1ea, "I");
+    if (cursorTickCount == 100) cursorTickCount = 0;
+
+    delete pos;
+    return;
+}
```

Comment:
Files must end with a newline, this is helpful to git, looks like your change deleted the last newline. (You can configure your editor for this)

Reply:
I'll avoid this mistake next time.